scripts/wallet.js

```diff
@@ -701,7 +701,29 @@ export class Wallet {
         // This is just to be sure since blockbook (as we know)
         // usually does not return txs of the actual last block.
         this.#lastProcessedBlock = blockCount - 5;
-        await this.#transparentSync();
+        // Transparent sync is inherently less stable than Shield since it requires heavy
+        // explorer indexing, so we'll attempt once asynchronously, and if it fails, set a
+        // recurring "background" sync interval until it's finally successful.
+        try {
+            await this.#transparentSync();
+        } catch {
+            // We'll set a 5s interval sync until it's finally successful, then nuke the 'thread'.
+            const cThread = new AsyncInterval(async () => {
```

Comment:
You cannot use `AsyncInterval`: the function `sync()` has to be blocked until `transparentSync()` is succesful, otherwise we end up in a state where the wallet is marked as synced, `this.#isSynced = true`, even if it isn't.

Reply:
The issue with blocking it is Shield then becomes unusable unless Transparent syncs (right now, Shield can be used even if Transparent is unable).

Reply:
Ideally we would split the state so that we have `transparentSynced` and `shieldSynced`, so they are operable independently.